internal/controller/notificationservice_controller.go

```diff
@@ -0,0 +1,127 @@
+// +kubebuilder:rbac:groups=pipelinesascode.tekton.dev,resources=repositories,verbs=get;list;watch
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the NotificationService object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.18.2/pkg/reconcile
+func (r *NotificationServiceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+
+	logger := r.Log.WithValues("pipelinerun", req.NamespacedName)
+	pipelineRun := &tektonv1.PipelineRun{}
+
+	err := r.Get(ctx, req.NamespacedName, pipelineRun)
+	if err != nil {
+		logger.Error(err, "Failed to get pipelineRun for", "req", req.NamespacedName)
+		fmt.Printf("Failed to get pipelineRun for req %s\n", req.NamespacedName)
+		if errors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+
+		return ctrl.Result{}, err
+	}
+	logger.Info("Reconciling PipelineRun", "Name", pipelineRun.Name)
+	fmt.Printf("Reconciling PipelineRun - Name %s\n", pipelineRun.Name)
+
+	if !IsFinalizerExistInPipelineRun(pipelineRun, NotificationPipelineRunFinalizer) &&
+		!IsAnnotationExistInPipelineRun(pipelineRun, NotificationPipelineRunAnnotation, NotificationPipelineRunAnnotationValue) {
+		err = AddFinalizerToPipelineRun(ctx, pipelineRun, r, NotificationPipelineRunFinalizer)
+		if err != nil {
+			fmt.Printf("Failed to add finalizer - %s\n", err)
+		}
+	}
+
+	if IsPipelineRunEndedSuccessfully(pipelineRun) &&
+		!IsAnnotationExistInPipelineRun(pipelineRun, NotificationPipelineRunAnnotation, NotificationPipelineRunAnnotationValue) {
+		results, err := GetResultsFromPipelineRun(pipelineRun)
+		if err != nil {
+			fmt.Printf("Failed to get results - %s", err)
+		} else {
+			fmt.Printf("Results are: %s\n", results)
+			err = AddAnnotationToPipelineRun(ctx, pipelineRun, r, NotificationPipelineRunAnnotation, NotificationPipelineRunAnnotationValue)
+			if err != nil {
+				fmt.Printf("Failed to add annotation - %s\n", err)
+			}
+		}
+	}
+
+	if IsPipelineRunEndedSuccessfully(pipelineRun) &&
+		IsAnnotationExistInPipelineRun(pipelineRun, NotificationPipelineRunAnnotation, NotificationPipelineRunAnnotationValue) {
+		err = RemoveFinalizerFromPipelineRun(ctx, pipelineRun, r, NotificationPipelineRunFinalizer)
+		if err != nil {
+			fmt.Printf("Failed to remove finalizer - %s\n", err)
+		}
+	}
+	return ctrl.Result{}, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *NotificationServiceReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&tektonv1.PipelineRun{}).
+		// WithEventFilter(predicate.Funcs{
```

Comment:
worth checking if you can use the predicates for checking if the pipelinerun was completed and if the it was triggered for a push event.